temporian/implementation/numpy/data/plotter.py

```diff
@@ -14,53 +14,206 @@
+
+
+@dataclass
+class Group:
+    """Features / timestaps that get plotted together."""
+
+    items: List[GroupItem]
+
+
+Groups = List[Group]
+
+
+def matplotlib_backend():
+    from temporian.implementation.numpy.data import plotter_matplotlib
+
+    return plotter_matplotlib.Plotter
+
+
+def bokeh_backend():
+    from temporian.implementation.numpy.data import plotter_bokeh
+
+    return plotter_bokeh.Plotter
+
+
+BACKENDS: Dict[str, Callable] = {
+    "matplotlib": matplotlib_backend,
+    "bokeh": bokeh_backend,
+}
+
+
+def error_message_import_backend(backend: str) -> str:
+    return (
+        f"Cannot plot with selected backend={backend}. Solutions: (1) Install"
+        f" {backend} e.g. 'pip install {backend}', or (2) use a different"
+        " plotting backen, for example with 'plot(..., backend=\"<other"
+        f' backend>"). The supported backends are: {list(BACKENDS.keys())}.'
+    )
+
+
+def build_groups(
+    evsets: InputEventSet,
+    features: Optional[Set[str]],
+    allow_list: bool = True,
+) -> Groups:
+    """Sort user inputs into groups of features to plot together."""
+
+    if isinstance(evsets, EventSet):
+        # Plot each feature individually
+        groups = []
+        for feature_idx, feature in enumerate(evsets.schema.features):
+            if features is not None and feature.name not in features:
+                continue
+            groups.append(Group([GroupItem(evsets, feature_idx)]))
+        if len(groups) == 0:
+            # Plot the timestamps
+            groups.append(Group([GroupItem(evsets, -1)]))
+        return groups
+
+    if isinstance(evsets, tuple):
+        # Plot all the event sets and their features together
+        group_items = []
+        for evset in evsets:
+            if not isinstance(evset, EventSet):
+                raise ValueError(
+                    f"Expecting tuple of eventsets. Got {type(evset)} instead."
+                )
+            plot_for_current_evtset = False
+            for feature_idx, feature in enumerate(evset.schema.features):
+                if features is not None and feature.name not in features:
+                    continue
+                group_items.append(GroupItem(evset, feature_idx))
+                plot_for_current_evtset = True
+            if not plot_for_current_evtset:
+                group_items.append(GroupItem(evset, -1))
+
+        return [Group(group_items)]
+
+    if allow_list and isinstance(evsets, list):
+        groups = []
+        for x in evsets:
+            groups.extend(build_groups(x, features, allow_list=False))
+        return groups
+    raise ValueError("Non supported evsets input")
 
 
-class Style(Enum):
-    """Plotting style."""
+def normalize_features(features: InputFeatures) -> Optional[Set[str]]:
+    """Normalizes the "features" argument of plot."""
+
+    if features is None:
+        return None
+    if isinstance(features, str):
+        return {features}
+    if isinstance(features, list):

```

Comment:
`if isinstance(features, (list, tuple))`

Reply:
We are not expecting for the features to be tuples